alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -291,7 +300,8 @@ object MonitorRunner : JobRunner, CoroutineScope, AbstractLifecycleComponent() {
 
             if (triggerService.isQueryLevelTriggerActionable(triggerCtx, triggerResult)) {
                 val actionCtx = triggerCtx.copy(error = monitorResult.error ?: triggerResult.error)
-                for (action in trigger.actions) {
+                var threshold = if (triggerMaxActions < trigger.actions.size && triggerMaxActions >= 0) triggerMaxActions else trigger.actions.size
```

Comment:
So we want to guarantee all the actions execute even if the number of actions are more than the max actions setting?